src/content/posts/2024/07/10/detecting-flaky-tests.mdx

```diff
@@ -0,0 +1,60 @@
+---
+title: "Tuist now supports detecting flaky tests"
+categories: ["Swift", "XcodeProj", "Xcode"]
+excerpt: "Tuist now supports detecting flaky tests. Learn how it works and how it can help you ship more reliable software"
+author: pepicrft
+---
+
+When developing software, it's standard practice to write tests to ensure that code behaves as expected. In the Apple ecosystem, developers have been using [XCTest](https://developer.apple.com/documentation/xctest) to write tests for their Swift and Objective-C codebases. Recently, [Swift Testing](https://developer.apple.com/xcode/swift-testing/) was introduced, addressing many of XCTest's shortcomings and advancing the state of the art in testing Swift code.
+
+## Understanding Flaky Tests
+
+If you've written tests before, you might know that sometimes tests fail even though the code is correct. These are often referred to as ["flaky tests"](https://docs.gitlab.com/ee/development/testing_guide/flaky_tests.html). In other words, flaky tests are not deterministic. They might fail due to race conditions or dependencies on the environment in which they run.
```

Comment:
nit: Flaky tests can very much signal that the code is `incorrect`, so maybe worth rephrasing the first sentence.